src/ua_parser/loaders.py

```diff
@@ -0,0 +1,120 @@
+__all__ = [
+    "load_builtins",
+    "load_data",
+    "load_yaml",
+    "MatchersData",
+    "UserAgentDict",
+    "OSDict",
+    "DeviceDict",
+]
+
+import io
+import os
+from typing import Callable, List, Optional, Tuple, Type, Union, TypedDict, Literal
+
+from .core import Matchers, UserAgentMatcher, OSMatcher, DeviceMatcher
+
+load: Optional[Callable]
+SafeLoader: Optional[Type]
+try:
+    from yaml import load, CSafeLoader as SafeLoader
+except ImportError:
+    try:
+        from yaml import load, SafeLoader
+    except ImportError:
+        load = SafeLoader = None
+
+
+class _RegexDict(TypedDict):
```

Comment:
These types should be moved closer to the function which needs them. They also need documenting.